ethereum/eip712/message.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2022 Evmos Foundation
+// This file is part of the Evmos Network packages.
+//
+// Evmos is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Evmos packages are distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Evmos packages. If not, see https://github.com/evmos/evmos/blob/main/LICENSE
+package eip712
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+)
+
+type eip712MessagePayload struct {
+	payload        gjson.Result
+	numPayloadMsgs int
+	message        map[string]interface{}
+}
+
+const (
+	PAYLOAD_MSGS = "msgs"
```

Comment:
Should I convert these to camelCase? @fedekunze

Reply:
I think we should do this refactor, the golangci-lint check is failing because of this

Reply:
yes, please convert them